### DISCUSSION TITLE
Companion/OSC - On Reload

### DISCUSSION THREAD

**Discussion by juzz197** [cpvalente/ontime#1119] [Discussion]
I've recently set up Companion to drive Ontime.

When I use Companion to trigger an event reload, this doesn't appear to trigger an 'On Load' OSC command.

Is this correct behaviour?  Personally I think it is, On Load and On Reload should be treated differently.  

If so, would it be possible to expose 'On Reload' as a trigger?

Many thanks!

**Comment by cpvalente** [comment]
Hi @juzz197 , thank you for this.

It seems like Ontime only triggers the loadEvent when a new event is loaded, meaning that reload would not do so.

The idea behind this was that, when a new event is armed (loaded), we could be updating other devices data and that work should not be re-done if the event did not change

This is consistent with your thoughts of reloading having to be different.

However, I fear that this division is mostly rational, but not practical. We may not want to complicate the application state unless it offers real value.
Meaning that, from a systems integrations perspective, unless the distinction is relevant we likely want to keep it simple with just making sure reloading re-triggers the load event

What do you think? Do you have an example workflow where reloading is relevant?
Could also do with some input here @lukestein @alex-Arc

**Reply by alex-Arc** [reply]
I am unsure what the correct way is
But I could imagine a loading a file on load and then starting playback on play
If the event is then reloaded it would be practical for that to also trigger the reloading of the file so it is also ready to start again

**Reply by cpvalente** [reply]
thank you @alex-Arc , in your example, there would be no difference between load and reload. So we likely dont need and want to introduce a new event

**Reply by juzz197** [reply]
I certainly can't think of an example where I'd need load/reload to be treated differently at the moment!  But having Reload and Load trigger the 'On Load' action would allow me reset other applications to the correct state to continue.

**Comment by alex-Arc** [comment]
@juzz197 this is part of the new beta https://github.com/cpvalente/ontime/releases/tag/v3.4.0-beta
If you have a change to try it out and see if it matches your expectations that would be appreciated

**Reply by juzz197** [reply]
All working as expected having tested it this evening - thank you!

Also liking the Projected Time in Countdown - hope to see this in other views soon.
